tinyagentos/lazy_backend_proxy.py

```
@@ -0,0 +1,272 @@
+                f"did not become healthy within {_COLD_START_TIMEOUT}s"
+            )
+
+    async def _stop_subprocess(self) -> None:
+        if self._proc is None or self._proc.poll() is not None:
+            self._proc = None
+            return
+        logger.info("lazy-proxy :%d → stopping backend", self._proxy_port)
+        if self._stop_cmd:
+            stop_proc: subprocess.Popen | None = None
+            try:
+                stop_proc = subprocess.Popen(
```

Ruff (0.15.14)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)